### PR TITLE
Consume invitations - Finishes mumuki/mumuki-platform#226

### LINKED ISSUE #792: Unify user model
Unify laboratory user model:
* first_name
* last_name

### PR DESCRIPTION
**Pull request by rodri042**
Related to https://github.com/mumuki/mumuki-office/pull/37

Fixes #767
Fixes #792 

The view is something like this:
invitations

### CODE REVIEW

**Comment by flbulgarelli** [comment]
UI is somewhat awful. We need to put some more work on it. @aguspina could you fix it?

File: app/models/invitation.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Invitation < ActiveRecord::Base
+  def self.find_by_slug(slug)
+    Invitation
+      .where(slug: slug)
+      .where(Invitation.arel_table[:expiration_date].gt(Date.today))
+      .take!
+  rescue RangeError
+    raise RecordNotFound.new('The invitation does not exist or it has expired')
```
Comment (flbulgarelli) [inline app/models/invitation.rb]: The problem of doing the fetching this way is that you can not distinguish between missing and expired invitations. This is an usability problem for the final user.

Comment (flbulgarelli) [inline app/models/invitation.rb]: If you add the invitation-organization relation - see my next comment - you can rewrite this method quite easy: 

```ruby
class Organization
      def active_invitation
           invitations.first('expiration_date < ?', Date.today)
      end
end
```

Then just do `Organization.current.active_invitation` to get the current invitation, and then something like `invitation.active?` that checks if the given invitation is the active of this organization.

Comment (flbulgarelli) [inline app/models/invitation.rb]: :warning: Dates do not take into consideration Timezones nor hours or minutes.  Use `Time.now` or `Time.zone.now`

File: lib/events.rb [hunk 2]
```diff
@@ -11,6 +11,10 @@
     Organization.update_from_json! payload.deep_symbolize_keys[:organization]
   end
 
+  event :InvitationCreated do |payload|
+    Invitation.create payload.deep_symbolize_keys[:invitation]
```
Comment (flbulgarelli) [inline lib/events.rb]: Use `create!`

Also, delegate this method into the invitation, as `import_from_json!`, like in - most - other models.

File: app/controllers/invitations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class InvitationsController < ApplicationController
+  before_action :authenticate!
+  before_action :set_user!
+  before_action :set_invitation!
+
+  def show
+    redirect_to_organization! if @user.student_of? @organization
```
Comment (rodri042) [inline app/controllers/invitations_controller.rb]: But, the link is `http://mumuki.io/join/1f3g-curso` and not `http://orga.mumuki.io/join/1f3g-curso`. I don't know if the *current organization* is the organization the user is trying to join.

Comment (flbulgarelli) [inline app/controllers/invitations_controller.rb]: But... why not using `http://orga.mumuki.io/join/1f3g-curso`? From my point of view it has much more sense.

Comment (rodri042) [inline app/controllers/invitations_controller.rb]: Yes, but the link will grow and have redundant info that it's already in the invitation. I was thinking in the student copying the link from a classroom board :stuck_out_tongue:

File: app/models/concerns/with_permissions.rb [hunk 4]
```diff
@@ -5,9 +5,17 @@ module WithPermissions
...
+  end
+
+  def student_of?(organization)
+    permissions.student? organization.slug
+  end
+
   #FIXME may be able to remove now
   def student?
-    permissions.student? Organization.slug
+    student_of? Organization
```
Comment (aguspina) [inline app/models/concerns/with_permissions.rb]: Besides this is correct, i think `Organization.current` is more expressibe

Comment (flbulgarelli) [inline app/models/concerns/with_permissions.rb]: :+1:

File: config/locales/en.yml [hunk 5]
```diff
@@ -15,6 +16,7 @@ en:
   chapter_number: Chapter %{number} -
   classroom: Classroom
   clear_console: Clear console
+  confirm: Confirmar
```
Comment (flbulgarelli) [inline config/locales/en.yml]: Confirm

**Comment by flbulgarelli** [comment]
Could you attach an screenshot @aguspina ?

**Comment by flbulgarelli** [comment]
Please fix the following: 

* The user profile image should be centered
* The flash message is too short
* Name and last name should be not be nullable if you belong to a private organization, but....
* ...you should not be asked to fulfill your name and surname if you only belong to a public organizations
* The welcome caption should not assume gender
* Form labels should be properly centered

**Comment by aguspina** [comment]
screen shot 2017-06-26 at 7 55 06 pm

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -34,10 +34,9 @@ def login_settings
   end
 
   def validate_user_profile!
-    unless current_user.profile_completed?
-      flash.notice = I18n.t :please_fill_profile_data
-      redirect_to user_path
-    end
+    return !current_user.has_accessible_organizations? || current_user.profile_completed?
```
Comment (flbulgarelli) [inline app/controllers/application_controller.rb]: Have you missed an if or something like that? This code has little sense

**Comment by aguspina** [comment]
@flbulgarelli Please review and merge

**Comment by aguspina** [comment]
@flbulgarelli Review now and merge

**Pull request merged**